scripts/compile_bundle.sh

```diff
@@ -26,35 +27,42 @@ if [ ! -f ${DPCPP_ENV} ]; then
     echo "DPC++ compiler environment ${DPCPP_ENV} doesn't seem to exist."
     exit 2
 fi
+source ${DPCPP_ENV}
```

Comment:
dpcpp_env and mkl_env is only needed when compiling ipex.

Reply:
Yes, i wondered if it would have any side effect on the other build steps? 

The reason here was to fail early on external dependencies. 
(It's really annoying to have a 2h build fail in the middle, especially when it always re-builds everything)

I also noticed that the ENV variables might get mixed up, when the script runs multiple times. It might be wise to reset all used ENV vars at the beginning? Should i add that?

Reply:
I'm not very clear why source command would fail. Existence of the source target is checked before it. Did you experience failures caused by source before or this is just in case?
Our internal compilation of pytorch and torch vision doesn't have dpcpp env and oneapi mkl activated. Sourcing them afterwards is to align with our internal process.
When this script is executed with `bash compile_bundle.sh`, all environment variables set inside this bash script will be discarded when the execution finishes. Why ENV variables get mixed up when the script runs for multiple times?

Reply:
the `source` command fails when `set -u` is active.

I will revert the change and see how to ensure a clean env variable scope. Even documented`bash compile_bundle.sh` is not a common practice. Everyone simply runs `./compile_bundle.sh` (including me).

Reply:
torch audio would not build without mkl, so i moved the script includes above that